### ISSUE TITLE
Simd.js Bug: Destination register type mismatch assertion while returning simd types from asm functions. 

### ISSUE THREAD

**Issue by arunetm** [Microsoft/ChakraCore#867]
This seems like a register allocation bug when nested calls are involved. The issue happens when creating/finding register for dstOpnd during the irbuilding phase of the SIMD128ReturnU4 opcode. There seem to be a mismatch in the return register type (Expecting U4 and finding I4)

**Repro Case**

>
```javascript
function asmModule(stdlib, imports) {
    "use asm";
    var i4 = stdlib.SIMD.Int32x4;
    var i4check = i4.check;
    var u4 = stdlib.SIMD.Uint32x4;
    var u4check = u4.check;
    function func1(a)
    {
        a = i4check(a);
        return a;
    }
    function func2()
    {
        var x = u4(-1, -2, -3, -4);
        var k = i4(-1, -2, -3, -4);;
        k = i4check(func1(k));
        return u4check(x);  //Deliberate use of x here
    }
    return {func1:func1, func2:func2};
}
var m = asmModule(this, {g1:SIMD.Float32x4(90934.2,123.9,419.39,449.0), g2:SIMD.Uint32x4(-1065353216, -1073741824,-1077936128, -1082130432)});
var ret2 = m.func2();

**Flags**
>x64_debug
>-DumpOnCrash  -bgjit-  -on:asmjs -testtrace:asmjs -simdjs -AsmJsStopOnError -maic:0 -lic:0


**Actual result**
>
Successfully compiled asm.js code
ASSERTION 30176: (G:\arun\git\chakra-310\core\lib\Backend\Sym.cpp, line 382) stackSym->GetType() == type
 Failure: (stackSym->GetType() == type)
dump filename C:\Users\b-arpuru\AppData\Local\Temp\\CH_30176_18664.dmp
FATAL ERROR: ch.exe failed due to exception code c0000420

**Comment by abchatra** [comment]
@arunetm  Can this issue be closed now?

**Comment by arunetm** [comment]
Yes. The fix had already been merged to Release/1.2
